docker/dev/find_fastest_maven_repo.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Define a list of mirrors without using Bash arrays.
+# c.f. https://stackoverflow.com/a/23930212/724872
+read -r -d '' MAVEN_REPOS <<EOM
+https://repo1.maven.org/maven2/
+https://maven.aliyun.com/repository/central
+EOM
+
+# Find the fastest maven repo using ping.
+SPEED=99999.9
+for i in $MAVEN_REPOS; do
```

Comment:
https://github.com/sql-machine-learning/sqlflow/blob/452a2ea714d0c3c2f357dd56f854eecbea0ede23/scripts/test/mysql_pod.yaml#L17-L23

duplicate in three shell scripts - `find_fastest_*.sh`.

How about we merge these files into a single file, say `find_fastest_resources.bash` that defines the following Bash functions:

```bash
function find_fastest_url()
function find_fastest_apt_source()  # calls find_fastest_url
function find_fastest_maven_repo() # calls find_fastest_url
function find_fastest_go_proxy()     # calls find_fastest_url
```

In the shell script that needs the above functions, we can write

```bash
source /find_fastest_resources.bash
sudo (find_fastest_apt_source > /etc/apt/sources)
```

Reply:
To do this, we need to change the domain name list

https://github.com/sql-machine-learning/sqlflow/blob/452a2ea714d0c3c2f357dd56f854eecbea0ede23/docker/dev/find_fastest_apt_source.sh#L19-L24

into a list of URLs.

Please be aware we must change the domain name list into URL list because various domain name might corresponds to various methods, some could be `http://` and some others could be `https://`.  For example

```
http://mirror.aliyun.com/ubuntu
```

and

```
http://archive.ubuntu.com/ubuntu
```